File: app/src/main/java/com/whatsapp/stickerimporter/Sticker.java
package com.whatsapp.stickerimporter;

import android.os.Parcel;
import android.os.Parcelable;

import java.util.List;

/**
 * Represents a single sticker in a sticker pack
 */
public class Sticker implements Parcelable {
    public String imageFileName;
    public List<String> emojis;
    public long size;

    public Sticker(String imageFileName, List<String> emojis) {
        this.imageFileName = imageFileName;
        this.emojis = emojis;
        this.size = 0;
    }

    protected Sticker(Parcel in) {
        imageFileName = in.readString();
        emojis = in.createStringArrayList();
        size = in.readLong();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(imageFileName);
        dest.writeStringList(emojis);
        dest.writeLong(size);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<Sticker> CREATOR = new Creator<Sticker>() {
        @Override
        public Sticker createFromParcel(Parcel in) {
            return new Sticker(in);
        }

        @Override
        public Sticker[] newArray(int size) {
            return new Sticker[size];
        }
    };
}

File: app/src/main/java/com/mystickers/whatsapp/Sticker.kt
package com.mystickers.whatsapp

import android.os.Parcel
import android.os.Parcelable

/**
 * Represents a single sticker within a sticker pack.
 */
data class Sticker(
    val imageFileName: String,
    val emojis: List<String>,
    val size: Long = 0
) : Parcelable {
    constructor(parcel: Parcel) : this(
        parcel.readString() ?: "",
        parcel.createStringArrayList() ?: emptyList(),
        parcel.readLong()
    )

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(imageFileName)
        parcel.writeStringList(emojis)
        parcel.writeLong(size)
    }

    override fun describeContents(): Int = 0

    companion object CREATOR : Parcelable.Creator<Sticker> {
        override fun createFromParcel(parcel: Parcel): Sticker = Sticker(parcel)
        override fun newArray(size: Int): Array<Sticker?> = arrayOfNulls(size)
    }
}
